added details of connecting azure account per issue 13018 and the exa…

cmwilson21 opened the pull request:
added details of connecting azure account per issue 13018 and the example solution text

<!--
Thank you for contributing to this project! You must fill out the information below before we can review this pull request. By explaining why you're making a change (or linking to a pull request) and what changes you've made, we can triage your pull request to the best possible team for review.

See our [CONTRIBUTING.md](/main/CONTRIBUTING.md) for information how to contribute.

For changes to content in [site policy](https://github.com/github/docs/tree/main/content/github/site-policy), see the [CONTRIBUTING guide in the site-policy repo](https://github.com/github/site-policy/blob/main/CONTRIBUTING.md).

We cannot accept changes to our translated content right now. See the [types-of-contributions.md](/main/contributing/types-of-contributions.md#earth_asia-translations) for more information.

Thanks again!
-->

### Why:
Updated description of how to connect an Azure account per the example text provided in the issue comments. 
Closes [issue link](https://github.com/github/docs/issues/13018)

<!--
- If there's an existing issue for your change, please link to it.
- If there's _not_ an existing issue, please open one first to make it more likely that this update will be accepted: https://github.com/github/docs/issues/new/choose. -->

### What's being changed:
Line under the Connecting your Azure subscription to your enterprise account heading was expanded to include more detailed information and a link to Microsoft's authentication flow.
<!-- Share artifacts of the changes, be they code snippets, GIFs or screenshots; whatever shares the most context. If you made changes to the `content` directory, a table will populate in a comment below with the staging and live article links -->

### Check off the following:

- [x ] I have reviewed my changes in staging (look for "Automatically generated comment" and click **Modified** to view your latest changes).
- [x ] For content changes, I have completed the [self-review checklist](https://github.com/github/docs/blob/main/contributing/self-review.md#self-review).

### Writer impact (This section is for GitHub staff members only):

- [ ] This pull request impacts the contribution experience
  - [ ] I have added the 'writer impact' label
  - [ ] I have added a description and/or a video demo of the changes below (e.g. a "before and after video")

<!-- Description of the writer impact here -->

cmwilson21 committed:
added details of connecting azure account per issue 13018 and the example solution text

ramyaparimi commented:
@cmwilson21 
Thanks so much for opening a PR! I'll get this triaged for review :zap:

lecoursen commented on content/billing/managing-billing-for-your-github-account/connecting-an-azure-subscription-to-your-enterprise.md:
```diff
@@ -17,7 +17,7 @@ After you connect an Azure subscription, you can also manage your spending limit
 ## Connecting your Azure subscription to your enterprise account
 
-To connect your Azure subscription, you must have owner permissions to the subscription.
+At the time of connection, GitHub’s Subscription Permission Validation will request read only access to display the list of available subscriptions. **To select an Azure subscription, you must have owner permissions to the subscription.** If the default tenant does not have the right permissions, you may need to specify a different tenant ID. Learn more about [Microsoft’s authentication flow](htt ...
```
Thanks @cmwilson21! I left a few style updates in the suggestion, but let's also move this line into a note at the end of step 9, which is when it will be most relevant to folks. Feel free to ping me after that's done for another review! ⚡ 

```suggestion
{% data variables.product.company_short %}'s Subscription Permission Validation requests read-only access to display the list of available subscriptions. To select an Azure subscription, you must have owner permissions to the subscription. If the default tenant does not have the right permissions, you may need to specify a different tenant ID. For more information, see [Microsoft identity platform and OAuth 2.0 authorization code flow](https://docs.microsoft.com/en-us/azure/active-directory/develop/v2-oauth2-auth-code-flow#request-an-authorization-code) in Microsoft Docs.
```

cmwilson21 committed:
Update content/billing/managing-billing-for-your-github-account/connecting-an-azure-subscription-to-your-enterprise.md

Co-authored-by: Laura Coursen <lecoursen@github.com>

cmwilson21 committed:
Update connecting-an-azure-subscription-to-your-enterprise.md

Reverted line 20 to read as it had before, moved my changes down to step 9 and used suggested changes.

cmwilson21 commented:
@lecoursen Thanks for the comment! I made the updates and moved it down to step 9.

cmwilson21 committed:
Update connecting-an-azure-subscription-to-your-enterprise.md

In step 9, added the changes inside parenthesis with a "Note:" tag prefixing it

lecoursen commented on content/billing/managing-billing-for-your-github-account/connecting-an-azure-subscription-to-your-enterprise.md:
```diff
@@ -26,7 +26,7 @@ To connect your Azure subscription, you must have owner permissions to the subsc
 1. To sign in to your Microsoft account, follow the prompts.
 1. Review the "Permissions requested" prompt. If you agree with the terms, click **Accept**.
-1. Under "Select a subscription", select the Azure Subscription ID that you want to connect to your enterprise.
+1. Under "Select a subscription", select the Azure Subscription ID that you want to connect to your enterprise. (Note: {% data variables.product.company_short %}'s Subscription Permission Validation requests read-only access to display the list of available subscriptions. To select an Azure subscription, you must have owner permissions to the subscription. If the default tenant does not have the  ...
```
```suggestion
1. Under "Select a subscription", select the Azure Subscription ID that you want to connect to your enterprise.

   {% note %}

   **Note:** {% data variables.product.company_short %}'s Subscription Permission Validation requests read-only access to display the list of available subscriptions. To select an Azure subscription, you must have owner permissions to the subscription. If the default tenant does not have the right permissions, you may need to specify a different tenant ID. For more information, see [Microsoft identity platform and OAuth 2.0 authorization code flow](https://docs.microsoft.com/en-us/azure/active-directory/develop/v2-oauth2-auth-code-flow#request-an-authorization-code) in Microsoft Docs.

   {% endnote %}
```

lecoursen reviewed (dismissed):
Thanks @cmwilson21! I'm just adding the markup for a note box and will merge this after I make sure it's all looking as expected in the preview environment. ⚡

lecoursen committed:
Add :nail_care:

lecoursen committed:
Merge branch 'main' into azure-subscriptions-issue-13018

cmwilson21 commented:
Thanks @lecoursen . I should have thought about the note box. I'll remember that for next time!

lecoursen reviewed (approved):
Looks great on preview!
